### PR TITLE
refactor: eslint & external types

### PR DESCRIPTION
**Pull request by AVVS**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "plugins": [
+    "@typescript-eslint"
+  ],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "rules": {
+    "semi": ["error", "never"],
```
Comment (BelkinaDasha) [inline .eslintrc.json:21]: I've noticed we now have semicolons in the ts interfaces, maybe we'll avoid it for cs consistensy? https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md

File: packages/plugin-logger/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "../../tsconfig.json",
+  "references": [{
+    "path": "../core/tsconfig.json"
```
[the remark this answers is not in the document]
Comment (AVVS) [reply packages/plugin-logger/tsconfig.json:4]: tsconfig.json is picked up by the editor and shows errors during development in the IDE, tsconfig.build.json is the actual build file thats used, so, yeah, both are needed. Notice references are different

File: packages/core/test/suites/graceful-shutdown.js [hunk 3]
```diff
@@ -35,10 +35,12 @@ class ChildServiceRunner {
     const { stderr, stdout } = subProcess;
 
     stderr.pipe(split2()).on('data', (line) => {
+      console.warn(line)
```
[the remark this answers is not in the document]
Comment (AVVS) [reply packages/core/test/suites/graceful-shutdown.js:38]: going to fix!

File: packages/plugin-logger/src/logger.ts [hunk 4]
```diff
@@ -134,9 +152,19 @@ export function attach(this: Microfleet & ValidatorPlugin, opts: Partial<LoggerC
...
 
-  service.log = pinoms({
+  this.log = pinoms({
     ...options,
     streams,
     name: applicationName || serviceName,
   })
 }
+
+declare module '@microfleet/core' {
```
[the remark this answers is not in the document]
Comment (AVVS) [reply packages/plugin-logger/src/logger.ts:162]: this is for module augmentation, it tells typescript that when we have '@microfleet/plugin-logger' imported '@microfleet/core' will have its properties augmented through declaration merging

**Pull request merged**